v2/internal/testcommon/samples_tester.go

```diff
@@ -0,0 +1,247 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package testcommon
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+	"reflect"
+	"strings"
+
+	"github.com/emirpasic/gods/maps/linkedhashmap"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/apimachinery/pkg/util/yaml"
+
+	resources "github.com/Azure/azure-service-operator/v2/api/resources/v1beta20200601"
+	"github.com/Azure/azure-service-operator/v2/internal/resolver"
+	"github.com/Azure/azure-service-operator/v2/pkg/genruntime"
+)
+
+const refsPackage = "refs"
+const samplesPath = "../../config/samples"
+
+// exclusions slice contains resources to exclude from test as these are either not an arm resource
```

Comment:
I think the webtest is the hidden link reference one, I might put that part of this comment down on (or near) the actual webtest line and explain it there, because "hidden link references" is something webtest specific, not actually some general or broad concept.